windows/test-app.ts

```diff
@@ -0,0 +1,300 @@
+    createDir(path.join(destPath, windowsDir, projDir, 'Bundle'));
+
+    const manifestFilePath = findClosestPathTo('app.json');
+
+    //Read path to resources from manifest and copy them
+    fs.readFile(manifestFilePath, async function (error, content) {
+        var resourcesPaths = JSON.parse(content.toString()).resources;
+        if (!resourcesPaths) {
+            throw new Error('No resources specified in app.json');
+        }
+        resourcesPaths = resourcesPaths.windows ? resourcesPaths.windows : resourcesPaths;
+        for (const resource of resourcesPaths) {
+            await copyAndReplaceAll(path.relative(path.dirname(manifestFilePath), resource), destPath, path.join(windowsDir, projDir, 'Bundle', path.basename(resource)));
+        }
+    });
+
+    const projectFilesReplacements: Record<string, any> = {
+        regExpPatternsToRemove: ['//\\sclang-format\\s(on|off)\\s'],
+
+        '<BundleContentRoot>.*</BundleContentRoot>': `<BundleContentRoot>${path.relative(path.join(destPath, windowsDir, projDir), path.join(windowsDir, projDir, 'Bundle'))}\\</BundleContentRoot>`,
+        '<ManifestRootPath>.*</ManifestRootPath>': `<ManifestRootPath>${path.relative(path.join(destPath, windowsDir, projDir), path.dirname(manifestFilePath))}\\</ManifestRootPath>`,
+        '<ReactNativeModulePath>.*</ReactNativeModulePath>': `<ReactNativeModulePath>${path.relative(path.join(destPath, windowsDir, projDir), reactNativeModulePath)}\\</ReactNativeModulePath>`,
+    };
+
+    const projectFilesMappings = [
+        {
+            from: path.join(srcRootPath, projDir, 'ReactTestApp.vcxproj'),
+            to: path.join(windowsDir, projDir, 'ReactTestApp.vcxproj'),
+        },
+        {
+            from: path.join(srcRootPath, projDir, 'PropertySheet.props'),
+            to: path.join(windowsDir, projDir, 'PropertySheet.props'),
+        },
+        {
+            from: path.join(srcRootPath, projDir, 'Package.appxmanifest'),
+            to: path.join(windowsDir, projDir, 'Package.appxmanifest'),
+        },
+        {
+            from: path.join(srcRootPath, projDir, 'ReactTestApp.vcxproj.filters'),
+            to: path.join(
+                windowsDir,
+                projDir,
+                'ReactTestApp.vcxproj.filters',
+            ),
+        },
+        {
+            from: path.join(srcRootPath, projDir, 'ReactTestApp_TemporaryKey.pfx'),
+            to: path.join(
+                windowsDir,
+                projDir,
+                'ReactTestApp_TemporaryKey.pfx',
+            ),
+        },
+        {
+            from: path.join(srcRootPath, projDir, 'packages.config'),
+            to: path.join(windowsDir, projDir, 'packages.config'),
+        },
+    ];
+
+    for (const mapping of projectFilesMappings) {
+        await copyAndReplace(
+            mapping.from,
+            destPath,
+            mapping.to,
+            projectFilesReplacements
+        );
+    }
```

Comment:
Do we really have to wait for these to finish before we can move on? Otherwise, I think we should fire and forget:
```suggestion
    projectFilesMappings.forEach(({ from, to }) => {
        copyAndReplace(from, destPath, to, projectFilesReplacements);
    });
```
You can also destructure `mapping` in the argument list to save some characters 😄